src/context-providers/display/display-provider.tsx

```diff
@@ -18,16 +19,35 @@ export const displayReducer = (
     action: TDisplayAction
 ) => {
     switch (action.type) {
+        case "set-modal": {
+            state.modals.push(action.payload);
```

Comment:
allow 1 modal for now

```suggestion
            state.modals = [action.payload];
```

Reply:
fixed in the last push